emacs/init.sls

```diff
@@ -1,2 +1,111 @@
-emacs:
-  pkg.installed
+{% from 'emacs/settings.sls' import emacs with context %}
+
+{%- if emacs.from_pkg == True %}
+install-emacs:
+  pkg.installed:
+    - pkgs:
+        - emacs
+{% else %}
+
+
+## Pull down a github release of the desired version
+{% with url = 'https://github.com/emacs-mirror/emacs/archive/%s.tar.gz'|format(emacs.name) %}
+emacs|fetch-release:
+  archive.extracted:
+    - name: {{ emacs.build_dir }}
+    - source: {{ url }}
+    - archive_format: tar
+    - source_hash: {{ emacs.hash }}
+    - user: root
+    - group: root
+    - if_missing: {{ emacs.build_dir }}/{{ emacs.name }}
+      
+  file.rename:
+    - name: {{ emacs.build_dir }}/{{ emacs.name }}
+    - source: {{ emacs.build_dir }}/emacs-{{ emacs.name }}
+    - force: true
+{% endwith %}
+
+
+emacs|dependencies:
```

Comment:
Let's modify this naming convention, the only reason I ask is because this is far outside the standard name, something like `emacs-dependencies` is fine, let's just avoid using the pipe since it's not used in any of the other formulas at this time!